datafusion/common/src/utils/mod.rs

```diff
@@ -321,89 +321,180 @@ pub fn longest_consecutive_prefix<T: Borrow<usize>>(
     count
 }
 
+/// Creates single element [`ListArray`], [`LargeListArray`] and
```

Comment:
The idea is to pull the wrapping code into this structure so it remains consistent across list types